### ISSUE TITLE
Is there specific ANN-SNN conversion examples using FS-neuron?

### ISSUE THREAD

**Issue by matianyun** [christophstoeckl/FS-neurons#8]
Dear author, 
I found the example code just replaces the activate function output with FS-neuron's approximate value, could you show any specific conversion examples?

**Comment by christophstoeckl** [comment]
The idea here is to make the conversion as easy as possible, ideally without having to rewrite the whole model. 
So the best approach is to simply override the build in tensorflow activation function with the FS-function. 

This way, a model can be converted with a minimal amount of code change needed.
For example, in the file `tf_models/models/official/r1/resnet/imagenet_main.py` the model has been converted using just these two lines of code:

```
from fs_coding import *
replace_relu_with_fs()
```

**Comment by matianyun** [comment]
Thank you, I got your idea. But what confused me is that fs_function just passes the approximate real value rather than spikes to the next layer, so how does this conversion realizes spikes transmission between layers?

**Comment by christophstoeckl** [comment]
Oh I see. 
As the next layer waits for all spikes of the previous layer to arrive anyway, this is the more convenient (and more efficient) way of implementing this functionality as it would be mathematically equivalent to sending the spikes to the next layer sequentially over time.

**Comment by matianyun** [comment]
Thank you for your reply, I agree that fs-function is more efficient to convert activate function into spikes, could you tell me how do you convert conv layer and full connection layer? and how do you implement the spikes generation and collection?

**Comment by christophstoeckl** [comment]
In the file `fs_coding.py` you can find spike generation on line 85 and spike collection on line 90. 
One does not have to change the fully connected or the conv layers, as they only represent the connectivity patterns of weights, which are the same in the spiking network. 
Hence, to convert a conv layer one only has to convert the activation function of the postsynaptic neurons.

**Comment by matianyun** [comment]
Thank you for your reply, I got it. But what about the generation of the spike of the input image, aka how do you encode the input image? And how do you collect spikes for final classification?

**Comment by christophstoeckl** [comment]
Images usually have three color channels, where each color channel has 8 bit. 
A pixel can therefore be perfectly implemented by three ReLU FS-neurons where the number of time steps K=8. 
This way an input image can be represented in the network without any information loss, hence it is not necessary to do this in a simulation given that it would be mathematically equivalent.

The output neurons could be thought of as neurons with a very high firing threshold, such that they do not fire at all. 
On neuromorphic hardware one would then read out the membrane potential of these neurons to obtain the logits.

**Comment by matianyun** [comment]
Excellent! I got it. Thank you!
